### ISSUE TITLE
Spearman weighted corr have different absolute values when the sign of the vector is reversed.

### ISSUE THREAD

**Issue by 10MO10** [ahmademad/wCorr#62]
Thank you very much for your great package.
I am sorry that my English is bad.

x<-c(1,2,3,4,5,6,7,8,9,10)
y1<-c(11.5,9.5,8.5,6.5,6.5,6.5,3,3,3.5,0.5)
y2<-c(-11.5, -9.5, -8.5, -6.5, -6.5, -6.5, -3, -3, -3.5, -0.5)
w<-c(1,3,3,3,3,1,2,4,1,3)
weightedCorr(x,y1,"Spearman", w, FALSE, FALSE)
[1] -0.9719387
weightedCorr(x,y2,"Spearman", w, FALSE, FALSE)
[1] 0.9649349

I suggest 
 "rnki <- t1 + (n+1)/(2*n)*t2"  ->  "rnki <- t1 + (t2+1)/2"
in wrank function.

For example, when t1=0, n=1, t2=5,
 "t1 + (n+1)/(2*n)*t2 -> 5"
 "t1 + (t2+1)/2 -> 3".

I think 3 is better than 5 because it is the median.
Thank you for reading.

**Comment by pdbailey0** [comment]
Here is a simpler case:
```
a <- 1:5
b <- c(1:4,5)
w <- c(1,1,1,1,2)
weightedCorr(a, b, "Spearman", w); weightedCorr(a,-1* b, "Spearman", w)
# [1] 0.9353851
# [1] -0.9334955
```
the weights do need to vary within the group that ties, and I can't find a case where four values are sufficient. So this is the MWE I found.

**Comment by pdbailey0** [comment]
@10MO10 I'm pushing a new version soon that fixes this. Heck of a delay, so I doubt you're still interested, but here we are. also, I appear to have arrived at a formula that is equal to your formulation but expressed slightly differently because of how I derived it. I don't know why it didn't work when I tested it earlier--presumably, I didn't deal with the final case correctly, since that's not in the for loop.

But now I want to know if there a reference for your formula? Also, who can I thank for pointing this out and proposing a fix?
